patch/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+    project: project::Metadata,
+    repo: &git2::Repository,
+) -> anyhow::Result<()> {
+    let head = repo.head()?;
+    let current_branch = head.shorthand().unwrap_or("HEAD (no branch)");
+
+    term::headline(&format!(
+        "🌱 Creating patch for {}.",
+        term::format::highlight(project.name)
+    ));
+
+    let master = repo
+        .resolve_reference_from_short_name(&format!("rad/{}", project.default_branch))?
+        .target();
+    let master_oid = master
+        .map(|h| format!("{:.7}", h.to_string()))
+        .unwrap_or_else(String::new);
+
+    let head_ref = head.target();
+    let head_oid = head_ref
+        .map(|h| format!("{:.7}", h.to_string()))
+        .unwrap_or_else(String::new);
+
+    term::info!(
+        "Proposing {} ({}) <= {} ({}).",
+        term::format::highlight(project.default_branch.clone()),
+        term::format::secondary(&master_oid),
+        term::format::highlight(&current_branch),
+        term::format::secondary(&head_oid),
+    );
+
+    let (ahead, behind) = repo.graph_ahead_behind(
+        head_ref.unwrap_or_else(Oid::zero),
+        master.unwrap_or_else(Oid::zero),
+    )?;
+    term::info!(
+        "This branch is {} commit(s) ahead, {} commit(s) behind {}.",
+        term::format::highlight(ahead),
+        term::format::highlight(behind),
+        term::format::highlight(project.default_branch)
+    );
+
+    let merge_base_ref = repo.merge_base(
+        master.unwrap_or_else(Oid::zero),
+        head_ref.unwrap_or_else(Oid::zero),
+    );
+
+    git::list_commits(repo, &merge_base_ref.unwrap(), &head_ref.unwrap(), true)?;
+    term::blank();
+
+    if term::confirm("View changes?") {
+        // TODO(erikli): Replace with repo.diff()
+        let workdir = repo
+            .workdir()
+            .ok_or_else(|| anyhow!("Could not get workdir current repository."))?;
+        let diff = git::git(workdir, ["diff", &master_oid, &head_oid])?;
+
+        term::Editor::new().edit(&diff)?;
+    }
+
+    if !term::confirm_with_default("Create patch using commit(s) above?", true) {
+        return Err(anyhow!("Canceled."));
+    }
+
+    let title: String = term::text_input("Title", None)?;
+    let description = match term::Editor::new().edit("").unwrap() {
+        Some(rv) => rv,
+        None => String::new(),
+    };
+    term::success!(
+        "{} {}",
+        term::format::tertiary_bold("Description".to_string()),
+        term::format::tertiary("·".to_string()),
+    );
+    term::markdown(&description);
+    term::blank();
+
+    if term::confirm_with_default("Submit using title and description?", true) {
+        term::blank();
+
+        let message = [title, description].join("\n");
+        patch::create(repo, &message, true)?;
+
+        if term::confirm_with_default("Sync to seed?", true) {
+            let sync_options = rad_sync::Options {
+                origin: None,
+                seed: None,
+                refs: rad_sync::Refs::DefaultBranch,
```

Comment:
I think we want to sync the current branch here, not the default branch, eg. say you're proposing a commit on `feature/1`, that's the branch you'd want to sync. Have you tested on non-default branches?